<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>578</width>
    <height>451</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QString常用功能</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_10">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>字符串截取</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="Btn_front">
              <property name="text">
               <string>front &amp;&amp; back</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Btn_left">
              <property name="text">
               <string>left &amp;&amp; right</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QPushButton" name="Btn_first">
              <property name="text">
               <string>first &amp;&amp; last</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Btn_section">
              <property name="text">
               <string>section*</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>存储相关</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QPushButton" name="Btn_empty">
              <property name="text">
               <string>isNull &amp;&amp; isEmpty</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Btn_resize">
              <property name="text">
               <string>resize</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QPushButton" name="Btn_size">
              <property name="text">
               <string>size &amp;&amp; length</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Btn_fill">
              <property name="text">
               <string>fill</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>搜索和判断</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QPushButton" name="Btn_index">
              <property name="text">
               <string>indexOf &amp;&amp; lastIndexOf</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Btn_contains">
              <property name="text">
               <string>contains</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QPushButton" name="Btn_endswith">
              <property name="text">
               <string>endsWith &amp;&amp; startsWith</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Btn_count">
              <property name="text">
               <string>count</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>转换或修改</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QPushButton" name="Btn_trimmed">
              <property name="text">
               <string>trimmed &amp;&amp; simplified</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Btn_insert">
              <property name="text">
               <string>insert</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QPushButton" name="Btn_remove">
              <property name="text">
               <string>remove</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Btn_repalce">
              <property name="text">
               <string>repalce</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>输入</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>str1</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="Cbox_str1">
              <item>
               <property name="text">
                <string>G:\Qt6Book\QtSamples</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>str2</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="Cbox_str2">
              <item>
               <property name="text">
                <string>\</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>参数</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="Spin_num"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>结果显示</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QPushButton" name="Btn_clear">
              <property name="text">
               <string>clear</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="plainTextEdit"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>Btn_clear</sender>
   <signal>clicked()</signal>
   <receiver>plainTextEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
